python/tskit/trees.py

```diff
@@ -6613,27 +6618,33 @@ def divergence(
         :ref:`span normalise <sec_stats_span_normalise>`,
         and :ref:`return value <sec_stats_output_format>`.
 
-        As a special case, an index ``(j, j)`` will compute the
-        :meth:`diversity <.TreeSequence.diversity>` of ``sample_set[j]``.
+        ..note ::
+            The two sample sets that are compared may contain some of the same sample
+            nodes. When carrying out all pairwise comparisons between sets, there is
+            therefore the possibility of pairing a sample with itself (contributing
+            a divergence of zero towards the average). Such comparisons are normally
+            *included* in the mean divergence measure. However, as a special
+            case, if a sample set is compared against itself by specifying an
+            index ``(j, j)``, only the pairs of *distinct* samples are counted, and
+            the resulting value is therefore equivalent to the
+            :meth:`diversity <.TreeSequence.diversity>` of ``sample_set[j]``.
```

Comment:
```suggestion
        ..note ::
            To avoid unexpected results, sample sets should be nonoverlapping,
            since comparisons of individuals to themselves are not removed when computing
            divergence between distinct sample sets. (However, specifying an index
            ``(j, j)`` computes the :meth:`diversity <.TreeSequence.diversity>` of ``sample_set[j]``,
            which removes self comparisons to provide an unbiased estimate.)
```

I'm trying to make this shorter and therefore less distracting, since in practice people aren't going to actually have overlapping sets.